src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ConfirmationService.kt

```diff
@@ -44,20 +42,19 @@ class ConfirmationService(
 
     val prisoner = prisonerSearchService.getPrisoner(confirmation.prisonNumber!!)
 
-    with(prisoner) {
-      return if (isCurrentPrisoner) throw ConflictException("Confirming arrival of active prisoner: '$prisonNumber' is not supported.")
-      else admitOffender(confirmation, prisonNumber)
+    return when {
+      prisoner.isCurrentPrisoner -> throw ConflictException("Confirming arrival of active prisoner: '$prisoner.prisonNumber' is not supported.")
+      else -> admitOffender(confirmation, prisoner.prisonNumber)
     }
   }
 
   fun confirmReturnFromCourt(moveId: String, confirmation: ConfirmCourtReturnRequest): ConfirmCourtReturnResponse {
-
     val prisoner = prisonerSearchService.getPrisoner(confirmation.prisonNumber)
 
-    return if (prisoner.isCurrentPrisoner)
-      returnFromCourt(moveId, prisoner.prisonNumber, confirmation)
-    else
-      throw IllegalArgumentException("Confirming court return of inactive prisoner: '${prisoner.prisonNumber}' is not supported.")
+    return when {
+      prisoner.isCurrentPrisoner -> returnFromCourt(moveId, prisoner.prisonNumber, confirmation)
+      else -> throw ConflictException("Confirming court return of inactive prisoner: '${prisoner.prisonNumber}' is not supported.")
```

Comment:
Now throwing a conflict exception when trying to process an inactive prisoner as a court return.

This now mirrors trying to create a new booking for an active prisoner